File: api/Futurem.Sourcing.Api/Entities/OrderProduct.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Futurem.Sourcing.Api.Entities;

public class OrderProduct : BaseEntity
{
    public long CustomerId { get; set; }
    public long SupplierId { get; set; }
    public long? SourceOrderProductId { get; set; }
    public long SourceCustomerOrderId { get; set; }

    [MaxLength(80)]
    public string SystemSku { get; set; } = string.Empty;

    [MaxLength(120)]
    public string? CustomerItemNo { get; set; }

    [MaxLength(120)]
    public string CustomerBarcode { get; set; } = string.Empty;

    [MaxLength(120)]
    public string? SupplierItemNo { get; set; }

    public string NameCn { get; set; } = string.Empty;
    public string? NameEn { get; set; }
    public string? NameEs { get; set; }
    public string? Specification { get; set; }
    public string? Color { get; set; }
    public string Unit { get; set; } = "PCS";

    [Column(TypeName = "decimal(18,2)")]
    public decimal PurchaseUnitPrice { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal SalesUnitPrice { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonQty { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonLengthCm { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonWidthCm { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonHeightCm { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonCbm { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonGwKg { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CartonNwKg { get; set; }

    public long ImporterProfileId { get; set; }
    public string ImporterSnapshotJson { get; set; } = "{}";
    public long LabelTemplateId { get; set; }
    public string LabelTemplateSnapshotJson { get; set; } = "{}";
    public long MarkTemplateId { get; set; }
    public string MarkTemplateSnapshotJson { get; set; } = "{}";

    [MaxLength(20)]
    public string BatchCode { get; set; } = string.Empty;

    public string Status { get; set; } = "draft";
    public DateTime? LockedAt { get; set; }
    public bool NeedsReview { get; set; }
}

File: web/src/types/orderProduct.ts
export type OrderProductImage = {
  id?: number
  orderProductId?: number
  imageUrl: string
  imageType: 'main' | 'detail' | 'package' | 'reference' | string
  sortNo: number
  fileName?: string | null
  contentType?: string | null
}

export type OrderProduct = {
  id: number
  customerId: number
  supplierId: number
  sourceOrderProductId?: number | null
  sourceCustomerOrderId: number
  systemSku: string
  customerItemNo?: string | null
  customerBarcode: string
  supplierItemNo?: string | null
  nameCn: string
  nameEn?: string | null
  nameEs?: string | null
  specification?: string | null
  color?: string | null
  unit: string
  purchaseUnitPrice: number
  salesUnitPrice: number
  cartonQty: number
  cartonLengthCm: number
  cartonWidthCm: number
  cartonHeightCm: number
  cartonCbm: number
  cartonGwKg: number
  cartonNwKg: number
  importerProfileId: number
  importerSnapshotJson: string
  labelTemplateId: number
  labelTemplateSnapshotJson: string
  markTemplateId: number
  markTemplateSnapshotJson: string
  batchCode: string
  status: string
  lockedAt?: string | null
  needsReview?: boolean
  remark?: string | null
}

export type OrderProductLine = {
  id?: number
  quantity: number
  cartons: number
  cartonQty: number
  totalCbm?: number
  totalGwKg?: number
  totalNwKg?: number
}

export type OrderProductDraft = Omit<Partial<OrderProduct>,
  'supplierId' |
  'purchaseUnitPrice' |
  'salesUnitPrice' |
  'cartonQty' |
  'cartonLengthCm' |
  'cartonWidthCm' |
  'cartonHeightCm' |
  'cartonGwKg' |
  'cartonNwKg'> & {
  customerOrderId: number
  supplierId: number | null
  nameCn: string
  purchaseUnitPrice: number
  salesUnitPrice: number
  quantity: number
  cartons: number
  cartonQty: number
  cartonLengthCm: number
  cartonWidthCm: number
  cartonHeightCm: number
  cartonGwKg: number
  cartonNwKg: number
  images: OrderProductImage[]
}

export type CustomerImporterProfile = {
  id: number
  customerId: number
  name: string
  companyName: string
  taxIdOrRfc?: string | null
  address: string
  contactName?: string | null
  phone?: string | null
  email?: string | null
  logoUrl?: string | null
  defaultOriginText: string
  defaultLabelTemplateId?: number | null
  defaultMarkTemplateId?: number | null
  isDefault: boolean
  status: string
  remark?: string | null
}

export type LabelMarkTemplate = {
  id: number
  code: string
  name: string
  documentType: string
  templateType: 'product_label' | 'carton_mark' | string
  customerId?: number | null
  importerProfileId?: number | null
  designerMode: 'fixed' | 'visual' | string
  language: string
  paperSize: string
  paperWidthMm?: number | null
  paperHeightMm?: number | null
  orientation: string
  layoutJson: string
  body: string
  isDefault: boolean
  status: string
  remark?: string | null
}
